packages/realm/src/Configuration.ts

```diff
@@ -33,20 +33,115 @@ import {
  */
 export type MigrationCallback = (oldRealm: Realm, newRealm: Realm) => void;
 
+/**
+ * This describes the different options used to create a {@link Realm} instance.
+ */
 export type BaseConfiguration = {
+  /**
+   * The path to the file where the Realm database should be stored. For synced Realms, a relative path
+   * is used together with app id and user id in order to avoid collisions with other apps or users.
+   * An absolute path is left untouched and on some platforms (iOS and Android) the app might not have
+   * permissions to create or open the file - permissions are not validated.
+   * If a relative path is specified, it is relative to @link{Realm.App.Configuration.baseFilePath}.
+   * @since 0.10.0
+   */
   path?: string;
+  /**
+   * Specifies all the object types in this Realm. **Required** when first creating a Realm at this `path`.
+   * If omitted, the schema will be read from the existing Realm file.
+   * @since 0.10.0
+   */
   schema?: (RealmObjectConstructor<AnyRealmObject> | ObjectSchema)[];
+  /**
+   * **Required** (and must be incremented) after changing the `schema`.
```

Comment:
I first read this quickly as a required field even though the TS type is optional. Perhaps this word rearrangement could help with that:
```suggestion
   * If changing the `schema`, this is **required** and must be incremented.
```